File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [-1] * (n + 1)

        def stairs(self, n, dp):
            if n <= 1:
                return 1
            if dp[n] != -1:
                return dp[n]
            dp[n] = stairs(self,n - 1, dp) + stairs(self,n - 2, dp)
            return dp[n]
        return stairs(self,n,dp)

File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function (n) {
    let count = 0
    let dp = new Array(n + 1).fill(-1)
    var again = function (n, dp) {
        if (n <= 1) return 1
        if (dp[n] != -1) return dp[n]
        dp[n] = again(n - 1, dp) + again(n - 2, dp)
        return dp[n]
    }
    return again(n, dp)
};